cloud/linode/loadbalancers.go

```diff
@@ -229,6 +229,90 @@ func (l *loadbalancers) EnsureLoadBalancer(ctx context.Context, clusterName stri
 	return lbStatus, nil
 }
 
+func (l *loadbalancers) getNodeBalancerDeviceId(ctx context.Context, firewallID, nbID, page int) (int, bool, error) {
+	devices, err := l.client.ListFirewallDevices(ctx, firewallID, &linodego.ListOptions{PageSize: 500, PageOptions: &linodego.PageOptions{Page: page}})
```

Comment:
I don't think you need to handle pagination manually here - linodego abstracts it from you

Reply:
Though the output is []linodego.FirewallDevice. The PageOptions is passed as an input parameter. The max. page size is 500. If there are more than 500 items, this might be useful. 

As discussed, for the sake of simplicity, this could be skipped with just a single call rather than recursive set of calls.